src/Integration/Persistence/FileBindingSerializer.cs

```diff
@@ -0,0 +1,230 @@
+/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Alm.Authentication;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarQube.Client.Helpers;
+
+namespace SonarLint.VisualStudio.Integration.Persistence
+{
+    internal abstract class FileBindingSerializer : ISolutionBindingSerializer
+    {
+        private readonly ICredentialStore credentialStore;
+        private readonly IFile fileWrapper;
+
+        protected readonly ISourceControlledFileSystem sccFileSystem;
+        protected readonly ILogger logger;
+
+        protected enum WriteMode
```

Comment:
The legacy mode file save is queued up along with the saves for the rulesets in each project.
However, the new connected mode only has a single file that we just want to write out immediately.